include/irr/asset/IVirtualTexture.h

```diff
@@ -0,0 +1,750 @@
+        }
+        return s;
+    }
+
+protected:
+    template <typename DSlayout_t>
+    core::smart_refctd_dynamic_array<typename DSlayout_t::SBinding> getDSlayoutBindings_internal(uint32_t _pgtBinding = 0u, uint32_t _fsamplersBinding = 1u, uint32_t _isamplersBinding = 2u, uint32_t _usamplersBinding = 3u) const
+    {
+        using retval_t = core::smart_refctd_dynamic_array<typename DSlayout_t::SBinding>;
+        auto retval = core::make_refctd_dynamic_array<retval_t>(1u+(getFloatViews().size()?1u:0u)+(getIntViews().size()?1u:0u)+(getUintViews().size()?1u:0u));
+        auto* bindings = retval->data();
+
+        auto fillBinding = [](auto& bnd, uint32_t _binding, uint32_t _count) {
+            bnd.binding = _binding;
+            bnd.count = _count;
+            bnd.stageFlags = asset::ISpecializedShader::ESS_FRAGMENT;
+            bnd.type = asset::EDT_COMBINED_IMAGE_SAMPLER;
+            bnd.samplers = nullptr; //samplers are left for user to specify at will
+        };
+
+        fillBinding(bindings[0], _pgtBinding, 1u);
+
+        uint32_t i = 1u;
+        if (getFloatViews().size())
+        {
+            fillBinding(bindings[i], _fsamplersBinding, getFloatViews().size());
+            ++i;
+        }
+        if (getIntViews().size())
+        {
+            fillBinding(bindings[i], _isamplersBinding, getIntViews().size());
+            ++i;
+        }
+        if (getUintViews().size())
+        {
+            fillBinding(bindings[i], _usamplersBinding, getUintViews().size());
+        }
+
+        return retval;
+    }
+    template <typename DS_t>
+    std::pair<core::smart_refctd_dynamic_array<typename DS_t::SWriteDescriptorSet>, core::smart_refctd_dynamic_array<typename DS_t::SDescriptorInfo>>
+        getDescriptorSetWrites_internal(DS_t* _dstSet, uint32_t _pgtBinding = 0u, uint32_t _fsamplersBinding = 1u, uint32_t _isamplersBinding = 2u, uint32_t _usamplersBinding = 3u) const
+    {
+        using writes_t = core::smart_refctd_dynamic_array<typename DS_t::SWriteDescriptorSet>;
+        using info_t = core::smart_refctd_dynamic_array<typename DS_t::SDescriptorInfo>;
+        using retval_t = std::pair<writes_t, info_t>;
+
+        const uint32_t writeCount = 1u+(getFloatViews().size()?1u:0u)+(getIntViews().size()?1u:0u)+(getUintViews().size()?1u:0u);
+        const uint32_t infoCount = 1u + getFloatViews().size() + getIntViews().size() + getUintViews().size();
+
+        auto writes_array = core::make_refctd_dynamic_array<writes_t>(writeCount);
+        auto* writes = writes_array->data();
+        auto info_array = core::make_refctd_dynamic_array<info_t>(infoCount);
+        auto* info = info_array->data();
+
+        writes[0].binding = _pgtBinding;
+        writes[0].arrayElement = 0u;
+        writes[0].count = 1u;
+        writes[0].descriptorType = EDT_COMBINED_IMAGE_SAMPLER;
+        writes[0].dstSet = _dstSet;
+        writes[0].info = info;
+        info[0].desc = createPageTableView();
```

Comment:
every single time I get the writes a new page table view is being created!

See https://github.com/buildaworldnet/IrrlichtBAW/pull/474/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.h&file-filters%5B%5D=No+extension&hide-deleted-files=true#r417876361